### ISSUE TITLE
Vertex does not support user supplied identifiers

### ISSUE THREAD

**Issue by humutkazan** [ArcadeData/arcadedb#1012]
### Discussed in https://github.com/ArcadeData/arcadedb/discussions/1007

<div type='discussions-op-text'>

<sup>Originally posted by **humutkazan** April  6, 2023</sup>
I am trying to migrate to arcadedb and want to use it with gremlin server. Got arcade db up and running on docker with this command

`docker run --rm -p 2480:2480 -p 2424:2424 -p 8182:8182 -e "arcadedb.server.defaultDatabases=Imported[root]{import:https://github.com/ArcadeData/arcadedb-datasets/raw/main/orientdb/OpenBeer.gz}" --env JAVA_OPTS="-Darcadedb.server.rootPassword=playwithdata -Darcadedb.server.plugins=Redis:com.arcadedb.redis.RedisProtocolPlugin,MongoDB:com.arcadedb.mongo.MongoDBProtocolPlugin,Postgres:com.arcadedb.postgres.PostgresProtocolPlugin,GremlinServer:com.arcadedb.server.gremlin.GremlinServerPlugin" arcadedata/arcadedb:latest`

When I try to run my code I got this error

`org.apache.tinkerpop.gremlin.driver.exception.ResponseException: Vertex does not support user supplied identifiers`

it shows the location of error and it is the code below (can provide more info if needed be)

```
        var traversal = g.addV(someLabel).property(T.id, id);

        traversal = addTraversalProperties(traversal, properties);

        traversal.as("label")
                .V(toVertexIds).as("equipment")
                .addE(EDGE_LABEL)
                .from("label")
                .to("equipment")
                .tail()
                .select("label")
                .id()
                .tryNext().orElseThrow(() -> new IllegalStateException(String.format("result: %s", queryResult)));
```

I tried to add `graph.set-vertex-id=true` into the gremlin-server.properties file inside docker but nothing changed.

Does anyone have any idea what I can do the solve this issue? Thanks
</div>

**Comment by humutkazan** [comment]
Same thing when I run it via server.bat file. And I don't know if this is relevant but it throws me this error every time

```
2023-04-07 12:10:10.001 INFO  [GremlinServer] Channel started at port 8182.Exception in thread "main" com.arcadedb.exception.DatabaseOperationException: Error on creating new database instance
        at com.arcadedb.database.EmbeddedDatabase.openInternal(EmbeddedDatabase.java:1679)
        at com.arcadedb.database.EmbeddedDatabase.open(EmbeddedDatabase.java:182)
        at com.arcadedb.database.DatabaseFactory.open(DatabaseFactory.java:76)
        at com.arcadedb.server.ArcadeDBServer.getDatabase(ArcadeDBServer.java:399)
        at com.arcadedb.server.ArcadeDBServer.getDatabase(ArcadeDBServer.java:283)
        at com.arcadedb.server.ArcadeDBServer.loadDatabases(ArcadeDBServer.java:421)
        at com.arcadedb.server.ArcadeDBServer.start(ArcadeDBServer.java:144)
        at com.arcadedb.server.ArcadeDBServer.main(ArcadeDBServer.java:92)
Caused by: com.arcadedb.utility.LockException: Database 'graph' is locked by another process (path=C:\Users\username\Desktop\arcadedb-23.3.1\.\databases\graph)
        at com.arcadedb.database.EmbeddedDatabase.lockDatabase(EmbeddedDatabase.java:1517)
        at com.arcadedb.database.EmbeddedDatabase.checkForRecovery(EmbeddedDatabase.java:1615)
        at com.arcadedb.database.EmbeddedDatabase.openInternal(EmbeddedDatabase.java:1657)
        ... 7 more
Caused by: java.nio.channels.OverlappingFileLockException
        at java.base/sun.nio.ch.FileLockTable.checkList(FileLockTable.java:229)
        at java.base/sun.nio.ch.FileLockTable.add(FileLockTable.java:123)
        at java.base/sun.nio.ch.FileChannelImpl.tryLock(FileChannelImpl.java:1154)
        at java.base/java.nio.channels.FileChannel.tryLock(FileChannel.java:1165)
        at com.arcadedb.database.EmbeddedDatabase.lockDatabase(EmbeddedDatabase.java:1498)
        ... 9 more
```

**Comment by lvca** [comment]
ArcadeDB provides its own ID as a Record ID. You can provide your ID as a normal attribute. You can define a unique index on it for fast retrieval.

About the last issue, an ArcadeDB server was running in the background, that's why the file was locked.

**Comment by humutkazan** [comment]
Hi @lvca,

Second issue is a problem right now because I can't restart arcadedb since I get this lock issue. I get this issue even for the first time I run arcadedb with gremlin server plugin right after I restarted my computer to absolutely sure there is no running arcadedb server in the background. What do you think I should do here? Been trying to solve this since yesterday but couldn't come up with a solution.

When I first started arcadedb with this config 
`set ARCADEDB_SETTINGS=-Darcadedb.server.plugins="GremlinServer:com.arcadedb.server.gremlin.GremlinServerPlugin", -Darcadedb.server.rootPassword=playwithdata`

It starts and throws this error like I said before. below is the whole log

```
2023-04-25 12:35:42.887 INFO  [ArcadeDBServer] <ArcadeDB_0> ArcadeDB Server v23.3.1 (build da609e411ccf28afeab01015ad6c2ecca1dadb3d/1680693032550/main) is starting up...
2023-04-25 12:35:42.896 INFO  [ArcadeDBServer] <ArcadeDB_0> Running on Windows 10 10.0 - OpenJDK 64-Bit Server VM Temurin-11.0.13+8
2023-04-25 12:35:42.896 INFO  [ArcadeDBServer] <ArcadeDB_0> Starting ArcadeDB Server in development mode with plugins [GremlinServer] ...
2023-04-25 12:35:42.911 INFO  [ArcadeDBServer] <ArcadeDB_0> - JMX Metrics Started...WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/C:/Users/z004f79z/Desktop/arcadedb-23.3.1/lib/arcadedb-gremlin-23.3.1.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

2023-04-25 12:35:44.237 INFO  [DefaultGraphManager] Graph [graph] was successfully configured via [./config/gremlin-server.properties].
2023-04-25 12:35:44.238 INFO  [ServerGremlinExecutor] Initialized Gremlin thread pool.  Threads in pool named with pattern gremlin-*
2023-04-25 12:35:44.248 INFO  [ServerGremlinExecutor] Initialized GremlinExecutor and preparing GremlinScriptEngines instances.
2023-04-25 12:35:44.781 INFO  [ServerGremlinExecutor] Initialized gremlin-groovy GremlinScriptEngine and registered metrics
2023-04-25 12:35:44.786 INFO  [ServerGremlinExecutor] A GraphTraversalSource is now bound to [g] with graphtraversalsource[arcadegraph[./databases/graph], standard]
2023-04-25 12:35:44.787 INFO  [ServerGremlinExecutor] A GraphTraversalSource is now bound to [graph] with graphtraversalsource[arcadegraph[./databases/graph], standard]
2023-04-25 12:35:44.801 INFO  [OpLoader] Adding the standard OpProcessor.
2023-04-25 12:35:44.985 INFO  [OpLoader] Adding the session OpProcessor.
2023-04-25 12:35:44.987 INFO  [OpLoader] Adding the traversal OpProcessor.
2023-04-25 12:35:45.015 INFO  [GremlinServer] Executing start up LifeCycleHook
2023-04-25 12:35:45.024 INFO  [GremlinServer] Executed once at startup of Gremlin Server.
2023-04-25 12:35:45.030 INFO  [GremlinServer] idleConnectionTimeout was set to 0 which resolves to 0 seconds when configuring this value - this feature will be disabled
2023-04-25 12:35:45.031 INFO  [GremlinServer] keepAliveInterval was set to 0 which resolves to 0 seconds when configuring this value - this feature will be disabled
2023-04-25 12:35:45.066 INFO  [AbstractChannelizer] Configured application/vnd.graphbinary-v1.0 with org.apache.tinkerpop.gremlin.driver.ser.GraphBinaryMessageSerializerV1
2023-04-25 12:35:45.086 INFO  [AbstractChannelizer] Configured application/vnd.gremlin-v3.0+json with org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0
2023-04-25 12:35:45.087 INFO  [AbstractChannelizer] Configured application/json with org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0
2023-04-25 12:35:45.092 INFO  [AbstractChannelizer] Configured application/vnd.gremlin-v2.0+json with org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV2d0
2023-04-25 12:35:45.093 INFO  [AbstractChannelizer] application/json already has org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0 configured - it will not be replaced by org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV2d0, change order of serialization configuration if this is not desired.
2023-04-25 12:35:45.805 INFO  [ArcadeDBServer] <ArcadeDB_0> - GremlinServer plugin started
2023-04-25 12:35:45.810 INFO  [GremlinServer] Gremlin Server configured with worker thread pool of 1, gremlin pool of 12 and boss thread pool of 1.
2023-04-25 12:35:45.811 INFO  [GremlinServer] Channel started at port 8182.Exception in thread "main" com.arcadedb.exception.DatabaseOperationException: Error on creating new database instance
        at com.arcadedb.database.EmbeddedDatabase.openInternal(EmbeddedDatabase.java:1679)
        at com.arcadedb.database.EmbeddedDatabase.open(EmbeddedDatabase.java:182)
        at com.arcadedb.database.DatabaseFactory.open(DatabaseFactory.java:76)
        at com.arcadedb.server.ArcadeDBServer.getDatabase(ArcadeDBServer.java:399)
        at com.arcadedb.server.ArcadeDBServer.getDatabase(ArcadeDBServer.java:283)
        at com.arcadedb.server.ArcadeDBServer.loadDatabases(ArcadeDBServer.java:421)
        at com.arcadedb.server.ArcadeDBServer.start(ArcadeDBServer.java:144)
        at com.arcadedb.server.ArcadeDBServer.main(ArcadeDBServer.java:92)
Caused by: com.arcadedb.utility.LockException: Database 'graph' is locked by another process (path=C:\Users\z004f79z\Desktop\arcadedb-23.3.1\.\databases\graph)
        at com.arcadedb.database.EmbeddedDatabase.lockDatabase(EmbeddedDatabase.java:1517)
        at com.arcadedb.database.EmbeddedDatabase.checkForRecovery(EmbeddedDatabase.java:1615)
        at com.arcadedb.database.EmbeddedDatabase.openInternal(EmbeddedDatabase.java:1657)
        ... 7 more
Caused by: java.nio.channels.OverlappingFileLockException
        at java.base/sun.nio.ch.FileLockTable.checkList(FileLockTable.java:229)
        at java.base/sun.nio.ch.FileLockTable.add(FileLockTable.java:123)
        at java.base/sun.nio.ch.FileChannelImpl.tryLock(FileChannelImpl.java:1154)
        at java.base/java.nio.channels.FileChannel.tryLock(FileChannel.java:1165)
        at com.arcadedb.database.EmbeddedDatabase.lockDatabase(EmbeddedDatabase.java:1498)
        ... 9 more
```

But everything works fine even though this error message. The problem is when I try to terminate this arcadedb process and restart it again to test persistency. When I restarted it again I get an error. below is the whole log

```
2023-04-25 12:33:45.549 INFO  [ArcadeDBServer] <ArcadeDB_0> ArcadeDB Server v23.3.1 (build da609e411ccf28afeab01015ad6c2ecca1dadb3d/1680693032550/main) is starting up...
2023-04-25 12:33:45.558 INFO  [ArcadeDBServer] <ArcadeDB_0> Running on Windows 10 10.0 - OpenJDK 64-Bit Server VM Temurin-11.0.13+8
2023-04-25 12:33:45.559 INFO  [ArcadeDBServer] <ArcadeDB_0> Starting ArcadeDB Server in development mode with plugins [GremlinServer] ...
2023-04-25 12:33:45.572 INFO  [ArcadeDBServer] <ArcadeDB_0> - JMX Metrics Started...Graph [graph] configured at [./config/gremlin-server.properties] could not be instantiated and will not be available in Gremlin Server.  GraphFactory message: GraphFactory could not instantiate this Graph implementation [class com.arcadedb.gremlin.ArcadeGraph]
java.lang.RuntimeException: GraphFactory could not instantiate this Graph implementation [class com.arcadedb.gremlin.ArcadeGraph]
        at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:84)
        at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:72)
        at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:106)
        at org.apache.tinkerpop.gremlin.server.util.DefaultGraphManager.addGraph(DefaultGraphManager.java:63)
        at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
        at org.apache.tinkerpop.gremlin.server.util.DefaultGraphManager.<init>(DefaultGraphManager.java:58)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor.<init>(ServerGremlinExecutor.java:86)
        at org.apache.tinkerpop.gremlin.server.GremlinServer.<init>(GremlinServer.java:127)
        at org.apache.tinkerpop.gremlin.server.GremlinServer.<init>(GremlinServer.java:90)
        at com.arcadedb.server.gremlin.GremlinServerPlugin.startService(GremlinServerPlugin.java:85)
        at com.arcadedb.server.ArcadeDBServer.registerPlugins(ArcadeDBServer.java:211)
        at com.arcadedb.server.ArcadeDBServer.start(ArcadeDBServer.java:141)
        at com.arcadedb.server.ArcadeDBServer.main(ArcadeDBServer.java:92)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.apache.tinkerpop.gremlin.structure.util.GraphFactory.open(GraphFactory.java:80)
        ... 16 more
Caused by: com.arcadedb.exception.DatabaseOperationException: Error on creating new database instance
        at com.arcadedb.database.EmbeddedDatabase.openInternal(EmbeddedDatabase.java:1679)
        at com.arcadedb.database.EmbeddedDatabase.open(EmbeddedDatabase.java:182)
        at com.arcadedb.database.DatabaseFactory.open(DatabaseFactory.java:76)
        at com.arcadedb.database.DatabaseFactory.open(DatabaseFactory.java:68)
        at com.arcadedb.gremlin.ArcadeGraph.<init>(ArcadeGraph.java:103)
        at com.arcadedb.gremlin.ArcadeGraph.open(ArcadeGraph.java:128)
        ... 21 more
Caused by: com.arcadedb.utility.LockException: Database './databases/graph' is locked by another process (path=C:\Users\z004f79z\Desktop\arcadedb-23.3.1\.\databases\graph)
        at com.arcadedb.database.EmbeddedDatabase.lockDatabase(EmbeddedDatabase.java:1517)
        at com.arcadedb.database.EmbeddedDatabase.checkForRecovery(EmbeddedDatabase.java:1615)
        at com.arcadedb.database.EmbeddedDatabase.openInternal(EmbeddedDatabase.java:1657)
        ... 26 more
Caused by: java.nio.channels.OverlappingFileLockException
        at java.base/sun.nio.ch.FileLockTable.checkList(FileLockTable.java:229)
        at java.base/sun.nio.ch.FileLockTable.add(FileLockTable.java:123)
        at java.base/sun.nio.ch.FileChannelImpl.tryLock(FileChannelImpl.java:1154)
        at java.base/java.nio.channels.FileChannel.tryLock(FileChannel.java:1165)
        at com.arcadedb.database.EmbeddedDatabase.lockDatabase(EmbeddedDatabase.java:1498)
        ... 28 more

2023-04-25 12:33:47.519 INFO  [ServerGremlinExecutor] Initialized Gremlin thread pool.  Threads in pool named with pattern gremlin-*
2023-04-25 12:33:47.532 INFO  [ServerGremlinExecutor] Initialized GremlinExecutor and preparing GremlinScriptEngines instances.WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/C:/Users/z004f79z/Desktop/arcadedb-23.3.1/lib/arcadedb-gremlin-23.3.1.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Could not create GremlinScriptEngine for gremlin-groovy
java.lang.IllegalStateException: javax.script.ScriptException: javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: graph for class: Script1
        at org.apache.tinkerpop.gremlin.jsr223.DefaultGremlinScriptEngineManager.lambda$createGremlinScriptEngine$16(DefaultGremlinScriptEngineManager.java:464)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:274)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
        at org.apache.tinkerpop.gremlin.jsr223.DefaultGremlinScriptEngineManager.createGremlinScriptEngine(DefaultGremlinScriptEngineManager.java:450)
        at org.apache.tinkerpop.gremlin.jsr223.DefaultGremlinScriptEngineManager.getEngineByName(DefaultGremlinScriptEngineManager.java:219)
        at org.apache.tinkerpop.gremlin.jsr223.CachedGremlinScriptEngineManager.lambda$getEngineByName$0(CachedGremlinScriptEngineManager.java:57)
        at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
        at org.apache.tinkerpop.gremlin.jsr223.CachedGremlinScriptEngineManager.getEngineByName(CachedGremlinScriptEngineManager.java:57)
        at org.apache.tinkerpop.gremlin.groovy.engine.GremlinExecutor.lambda$eval$0(GremlinExecutor.java:272)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: javax.script.ScriptException: javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: graph for class: Script1
        at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:383)
        at java.scripting/javax.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:264)
        at org.apache.tinkerpop.gremlin.jsr223.DefaultGremlinScriptEngineManager.lambda$createGremlinScriptEngine$16(DefaultGremlinScriptEngineManager.java:460)
        ... 24 more
Caused by: javax.script.ScriptException: groovy.lang.MissingPropertyException: No such property: graph for class: Script1
        at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:702)
        at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:381)
        ... 26 more
Caused by: groovy.lang.MissingPropertyException: No such property: graph for class: Script1
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:65)
        at org.codehaus.groovy.runtime.callsite.PogoGetPropertySite.getProperty(PogoGetPropertySite.java:51)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGroovyObjectGetProperty(AbstractCallSite.java:309)
        at Script1.run(Script1.groovy:32)
        at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:679)
        ... 27 more
Could not initialize gremlin-groovy GremlinScriptEngine as init script could not be evaluated
java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: gremlin-groovy is not an available GremlinScriptEngine
        at java.base/java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:412)
        at java.base/java.util.concurrent.CompletableFuture.join(CompletableFuture.java:2044)
        at org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor.lambda$new$4(ServerGremlinExecutor.java:153)
        at java.base/java.util.LinkedHashMap$LinkedKeySet.forEach(LinkedHashMap.java:559)
        at org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor.<init>(ServerGremlinExecutor.java:145)
        at org.apache.tinkerpop.gremlin.server.GremlinServer.<init>(GremlinServer.java:127)
        at org.apache.tinkerpop.gremlin.server.GremlinServer.<init>(GremlinServer.java:90)
        at com.arcadedb.server.gremlin.GremlinServerPlugin.startService(GremlinServerPlugin.java:85)
        at com.arcadedb.server.ArcadeDBServer.registerPlugins(ArcadeDBServer.java:211)
        at com.arcadedb.server.ArcadeDBServer.start(ArcadeDBServer.java:141)
        at com.arcadedb.server.ArcadeDBServer.main(ArcadeDBServer.java:92)
Caused by: java.lang.IllegalArgumentException: gremlin-groovy is not an available GremlinScriptEngine
        at org.apache.tinkerpop.gremlin.jsr223.CachedGremlinScriptEngineManager.registerLookUpInfo(CachedGremlinScriptEngineManager.java:95)
        at org.apache.tinkerpop.gremlin.jsr223.CachedGremlinScriptEngineManager.getEngineByName(CachedGremlinScriptEngineManager.java:58)
        at org.apache.tinkerpop.gremlin.groovy.engine.GremlinExecutor.lambda$eval$0(GremlinExecutor.java:272)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)

2023-04-25 12:33:48.398 INFO  [OpLoader] Adding the standard OpProcessor.
2023-04-25 12:33:48.569 INFO  [OpLoader] Adding the session OpProcessor.
2023-04-25 12:33:48.571 INFO  [OpLoader] Adding the traversal OpProcessor.
2023-04-25 12:33:48.598 INFO  [GremlinServer] idleConnectionTimeout was set to 0 which resolves to 0 seconds when configuring this value - this feature will be disabled
2023-04-25 12:33:48.599 INFO  [GremlinServer] keepAliveInterval was set to 0 which resolves to 0 seconds when configuring this value - this feature will be disabled
2023-04-25 12:33:48.629 INFO  [AbstractChannelizer] Configured application/vnd.graphbinary-v1.0 with org.apache.tinkerpop.gremlin.driver.ser.GraphBinaryMessageSerializerV1
2023-04-25 12:33:48.648 INFO  [AbstractChannelizer] Configured application/vnd.gremlin-v3.0+json with org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0
2023-04-25 12:33:48.649 INFO  [AbstractChannelizer] Configured application/json with org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0
2023-04-25 12:33:48.654 INFO  [AbstractChannelizer] Configured application/vnd.gremlin-v2.0+json with org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV2d0
2023-04-25 12:33:48.654 INFO  [AbstractChannelizer] application/json already has org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0 configured - it will not be replaced by org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV2d0, change order of serialization configuration if this is not desired.
2023-04-25 12:33:49.341 INFO  [ArcadeDBServer] <ArcadeDB_0> - GremlinServer plugin started
2023-04-25 12:33:49.343 INFO  [HttpServer] <ArcadeDB_0> - Starting HTTP Server (host=0.0.0.0 port=2480-2489 httpsPort=2490-2499)...
2023-04-25 12:33:49.345 INFO  [GremlinServer] Gremlin Server configured with worker thread pool of 1, gremlin pool of 12 and boss thread pool of 1.
2023-04-25 12:33:49.346 INFO  [GremlinServer] Channel started at port 8182.
2023-04-25 12:33:49.378 INFO  [undertow] starting server: Undertow - 2.3.5.Final
2023-04-25 12:33:49.387 INFO  [xnio] XNIO version 3.8.8.Final
2023-04-25 12:33:49.394 INFO  [nio] XNIO NIO Implementation Version 3.8.8.Final
2023-04-25 12:33:49.415 INFO  [threads] JBoss Threads version 3.5.0.Final
2023-04-25 12:33:49.454 INFO  [HttpServer] <ArcadeDB_0> - HTTP Server started (host=0.0.0.0 port=2480 httpsPort=2490)
2023-04-25 12:33:49.467 INFO  [ArcadeDBServer] <ArcadeDB_0> Available query languages: [sqlscript, mongo, gremlin, java, cypher, js, graphql, sql]
2023-04-25 12:33:49.469 INFO  [ArcadeDBServer] <ArcadeDB_0> ArcadeDB Server started in 'development' mode (CPUs=12 MAXRAM=7,95GB)
2023-04-25 12:33:49.470 INFO  [ArcadeDBServer] <ArcadeDB_0> Studio web tool available at http://172.22.16.1:2480
```

Should I use arcadedb with external gremlin server instead of as arcadedb plugin or am I missing some important config here? 
Thanks in advance
